### PR TITLE
feat: LT-1121 - Super Users and Administer Role Permission

### PR DESCRIPTION
**Pull request by cstobart**

### CODE REVIEW

File: templates/users/index.html [hunk 1]
```diff
@@ -55,7 +55,11 @@ <h2 class="govuk-heading-l">{{ title }}</h2>
                     <td class="govuk-table__cell">{{ user.role.name }}</td>
                     <td class="govuk-table__cell">{{ user.status }}</td>
                     <td class="govuk-table__cell">
-                        <a href="{% url 'users:edit' user.id %}" style="margin: 0;" class="govuk-link govuk-link--no-visited-state">Edit</a>
+                        {% if super_user %}
+                            <a href="{% url 'users:edit' user.id %}" style="margin: 0;" class="govuk-link govuk-link--no-visited-state">Edit</a>
```
Comment (jfaracik) [inline templates/users/index.html]: Use {% lcs 'CONSTANT HERE' %} for strings

File: templates/users/profile.html [hunk 2]
```diff
@@ -15,14 +15,20 @@ <h2 class="govuk-heading-l">
...
-				{% endif %}
-			{% endif %}
+            {% if super_user %}
+                <a href="{% url 'users:edit' data.user.id %}" type="submit" class="govuk-button">Edit</a>
+            {% elif not data.user.role.name == 'Super User' %}
+                <a href="{% url 'users:edit' data.user.id %}" type="submit" class="govuk-button">Edit</a>
+            {% endif %}
+            {% if not data.user.role.name == 'Super User' %}
+                {% if request.user.lite_api_user_id|stringformat:"s" != data.user.id|stringformat:"s" %}
+                    {% if data.user.status == 'Active' %}
```
Comment (jfaracik) [inline templates/users/profile.html]: And LCS all here too!

File: templates/users/roles.html [hunk 3]
```diff
@@ -15,11 +15,13 @@ <h2 class="govuk-heading-l">{{ title }}</h2>
...
 		</div>
-		<div class="lite-app-bar__controls">
-			<a id="button_add_role" href="{% url 'users:add_role' %}" class="govuk-button" draggable="false" role="button">
-				{% get_string 'roles.add.title' %}
-			</a>
-		</div>
+        {% if 'ADMINISTER_ROLES' in user_permissions %}
+            <div class="lite-app-bar__controls">
+                <a id="button_add_role" href="{% url 'users:add_role' %}" class="govuk-button" draggable="false" role="button">
+                    {% get_string 'roles.add.title' %}
```
Comment (jfaracik) [inline templates/users/roles.html]: LCS baby

File: templates/users/roles.html [hunk 4]
```diff
@@ -58,7 +60,11 @@ <h2 class="govuk-heading-l">{{ title }}</h2>
 						</ul>
 					</td>
                     <td class="govuk-table__cell">
-						<a href="{% url 'users:edit_role' role.id %}" class="govuk-link govuk-link--no-visited-state">Edit</a>
+                        {% if not role.id == '00000000-0000-0000-0000-000000000002' %}
```
Comment (jfaracik) [inline templates/users/roles.html]: Don't feel too comfortable hardcoding UUIDs in HTML. Is there a better way?

File: users/views/users.py [hunk 5]
```diff
@@ -37,10 +40,13 @@ class ViewUser(TemplateView):
     def get(self, request, **kwargs):
         data, _ = get_gov_user(request, str(kwargs['pk']))
         user = data.get('user')
+        request_user, _ = get_gov_user(request, str(request.user.lite_api_user_id))
+        super_user = request_user['user']['role']['name'] == 'Super User'
 
         context = {
             'data': data,
-            'title': user.get('first_name') + ' ' + user.get('last_name')
+            'title': user.get('first_name') + ' ' + user.get('last_name'),
```
Comment (jfaracik) [inline users/views/users.py]: Don't put title in context, instead put it in the HTML like so

<h2 class="govuk-heading-l">{% block title %}Title goes here!{% endblock %}</h2>

File: templates/users/roles.html (hunk 4 above)
Comment (cstobart) [reply templates/users/roles.html]: Could set a condition in the context

File: templates/users/profile.html [hunk 6]
```diff
@@ -11,18 +11,28 @@
...
+                </a>
+            {% elif not data.user.role.name == 'Super User' %}
+                <a href="{% url 'users:edit' data.user.id %}" type="submit" class="govuk-button">
+                    {% lcs 'USER_PROFILE_PAGE_EDIT' %}
+                </a>
+            {% endif %}
+            {% if not data.user.role.name == 'Super User' %}
+                {% if request.user.lite_api_user_id|stringformat:"s" != data.user.id|stringformat:"s" %}
+                    {% if data.user.status == 'Active' %}
+                        <a href="{% url 'users:change_status' data.user.id 'deactivate' %}" role="button" draggable="false" class="govuk-button govuk-button--warning govuk-!-margin-right-2">Deactivate</a>
```
Comment (Mark-Spiers-kainos) [inline templates/users/profile.html]: move button text into constants please

File: templates/users/profile.html [hunk 7]
```diff
@@ -11,18 +11,32 @@
...
-					<a href="{% url 'users:change_status' data.user.id 'deactivate' %}" role="button" draggable="false" class="govuk-button govuk-button--warning govuk-!-margin-right-2">Deactivate</a>
-				{% elif data.user.status == 'Deactivated' %}
-					<a href="{% url 'users:change_status' data.user.id 'reactivate' %}" role="button" draggable="false" class="govuk-button govuk-!-margin-right-2">Reactivate</a>
-				{% endif %}
-			{% endif %}
+            {% if super_user %}
+                <a href="{% url 'users:edit' data.user.id %}" type="submit" class="govuk-button">
+                    {% lcs 'USER_PROFILE_PAGE_EDIT' %}
+                </a>
+            {% elif not data.user.role.name == 'Super User' %}
```
Comment (jfaracik) [inline templates/users/profile.html]: Refer to role id, not name

File: users/views/users.py [hunk 8]
```diff
@@ -54,13 +60,15 @@ def get(self, request, **kwargs):
 class EditUser(TemplateView):
     def get(self, request, **kwargs):
         user, _ = get_gov_user(request, str(kwargs['pk']))
-        return form_page(request, edit_user_form(request, str(kwargs['pk'])), data=user['user'])
+        super_user = user['user']['role']['name'] == 'Super User' and request.user.lite_api_user_id == str(kwargs['pk'])
```
Comment (jfaracik) [inline users/views/users.py]: Check the ID of the role, not the name

File: users/views/users.py (hunk 5 above)
Comment (jfaracik) [inline users/views/users.py]: Remove title from context as its already in HTML

File: users/views/users.py [hunk 9]
```diff
@@ -36,11 +40,15 @@ def post(self, request, **kwargs):
 class ViewUser(TemplateView):
     def get(self, request, **kwargs):
         data, _ = get_gov_user(request, str(kwargs['pk']))
-        user = data.get('user')
+        request_user, _ = get_gov_user(request, str(request.user.lite_api_user_id))
+        super_user = request_user['user']['role']['name'] == 'Super User'
```
Comment (jfaracik) [inline users/views/users.py]: Name here too, should be id

File: users/views/users.py [hunk 10]
```diff
@@ -2,20 +2,24 @@
...
+from lite_forms.generators import form_page
 from users.forms.users import add_user_form, edit_user_form
 from users.services import get_gov_users, post_gov_users, put_gov_user, get_gov_user
 
 
 class UsersList(TemplateView):
     def get(self, request, **kwargs):
         data, _ = get_gov_users(request)
+        user, _ = get_gov_user(request, str(request.user.lite_api_user_id))
+        super_user = user['user']['role']['name'] == 'Super User'
```
Comment (jfaracik) [inline users/views/users.py]: And here, could this not be a reusable function instead?

File: users/views/roles.py [hunk 11]
```diff
@@ -13,11 +15,14 @@ class Roles(TemplateView):
...
         roles, _ = get_roles(request)
         all_permissions, _ = get_permissions(request)
+        permissions = get_user_permissions(request)
 
         context = {
             'all_permissions': all_permissions['permissions'],
             'roles': roles['roles'],
             'title': get_string('roles.title'),
+            'user_permissions': permissions,
+            'super_user_role_id': '00000000-0000-0000-0000-000000000002'
```
Comment (jfaracik) [inline users/views/roles.py]: Pull from constants here

File: users/views/users.py (hunk 10 above)
Comment (jfaracik) [reply users/views/users.py]: This ***should*** be a function. 

`def is_superuser(user):
   return user['user']['role']['id'] == SUPER_USER_ROLE_ID`

File: users/forms/users.py [hunk 12]
```diff
@@ -21,17 +22,17 @@ def add_user_form(request):
...
                               name='email'),
                     Select(name='team',
                            title='What team will the user belong to?',
                            options=get_teams(request, True)),
-                    Select(name='role',
-                           options=get_roles(request, True),
-                           title='What role should this user have?'),
+                    conditional(not super_user, Select(name='role',
+                                                       options=get_roles(request, True),
+                                                       title='What role should this user have?'), )
```
Comment (jfaracik) [inline users/forms/users.py]: Strings!

**Pull request merged**